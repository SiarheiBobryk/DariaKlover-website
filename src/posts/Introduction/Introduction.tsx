import * as React from 'react';

import { OverridableStringUnion } from '@mui/types';
import Typography from '@mui/material/Typography';
import Link from '@mui/material/Link';
import OpenInNewIcon from '@mui/icons-material/OpenInNew';
// eslint-disable-next-line no-restricted-imports
import { Variant } from '@mui/material/styles/createTypography';

interface IntroductionProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  component?: React.ElementType<any>;
  variant?: OverridableStringUnion<Variant | 'inherit'>;
}

function Introduction(props: IntroductionProps) {
  const { component = 'h2', variant = 'h5' } = props;
  return (
    <>
      <Typography component={component} variant={variant}>
        Кто я и что я делаю
      </Typography>
      <Typography>Всем привет, я - Дарья Кловер!</Typography>
      <Typography>
        Я прохожу обучение в{' '}
        <Link
          href="https://uom.education/university"
          sx={{ display: 'inline-flex', alignItems: 'center', marginBottom: 0 }}
          rel="noopener"
        >
          Университете образовательной медицины (УОМ) <OpenInNewIcon fontSize="small" />
        </Link>{' '}
        и <i>PreventAge LifeStyle School</i> на интегративного health-коуча - специалиста по модификации образа жизни и
        немедикаментозному оздоровлению.
      </Typography>
      <Typography>
        Я помогаю своим клиентам вести здоровый образ жизни и предотвратить появление хронических заболеваний, а также
        повысить качество жизни, улучшить самочувствие, делюсь рекомендациями для восстановления и поддержания здоровья.
      </Typography>

      <Typography>
        Я разбираю запрос клиента с разных сторон, что помогает увидеть полную картину конкретной ситуации и ее причины.
        Я использую различные подходы и инструменты, чтобы помочь проработать эти причины и придти к решению запроса
        клиента.
      </Typography>

      <Typography>
        Одна из моих главных задач - осознание клиентом того, <strong>что, как и почему нужно делать</strong>, чтобы
        достичь желаемого результата.
      </Typography>
    </>
  );
}

export default Introduction;
