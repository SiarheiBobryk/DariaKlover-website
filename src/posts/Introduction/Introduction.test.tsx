import * as React from 'react';

import { render, RenderResult } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { axe, toHaveNoViolations } from 'jest-axe';

import { typographyClasses } from '@mui/material/Typography';

import Introduction from './Introduction';

expect.extend(toHaveNoViolations);

describe('<Introduction />', () => {
  describe('a11y', () => {
    it('should match a11y standards', async () => {
      const { container }: RenderResult = render(<Introduction />);

      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });

  describe('snapshotting', () => {
    it('should render correctly', () => {
      const introduction: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
        .create(<Introduction />)
        .toJSON();
      expect(introduction).toMatchSnapshot();
    });
  });

  describe('Elements inspection', () => {
    it('should have the `h2` HTML element as heading', () => {
      render(<Introduction />);

      const h2: HTMLHeadingElement | null = document.querySelector('h2');
      expect(h2).toBeInTheDocument();
    });

    it('should have the CSS classes from `h5` Typography variant by default', () => {
      render(<Introduction />);

      const h2: HTMLHeadingElement | null = document.querySelector('h2');
      expect(h2).toHaveClass(typographyClasses.h5);
    });

    it('should have a link with the right attributes', () => {
      render(<Introduction />);

      const link: HTMLAnchorElement | null = document.querySelector('p > a');
      expect(link).toBeInTheDocument();
      expect(link).toHaveTextContent('Университете образовательной медицины (УОМ)');
      expect(link).toHaveAttribute('href', 'https://uom.education/university');
      expect(link).toHaveAttribute('rel', 'noopener');
    });

    it('should have the open-in-new-window icon`', () => {
      render(<Introduction />);

      const icon: SVGElement | null = document.querySelector('a[href="https://uom.education/university"] > svg');
      expect(icon).toBeInTheDocument();
    });

    it('should have an italic (the `i` HTML element) school name', () => {
      render(<Introduction />);

      const i: HTMLElement | null = document.querySelector('i');
      expect(i).toHaveTextContent('PreventAge LifeStyle School');
    });

    it('should have the right `strong` element', () => {
      render(<Introduction />);

      const strong: HTMLElement | null = document.querySelector('strong');
      expect(strong).toHaveTextContent('что, как и почему нужно делать');
    });
  });
});
