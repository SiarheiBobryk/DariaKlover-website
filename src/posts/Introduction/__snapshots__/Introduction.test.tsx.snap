// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduction /> snapshotting should render correctly 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
  >
    Кто я и что я делаю
  </h2>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Всем привет, я - Дарья Кловер!
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Я прохожу обучение в
     
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="https://uom.education/university"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener"
    >
      Университете образовательной медицины (УОМ)
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
     
    и 
    <i>
      PreventAge LifeStyle School
    </i>
     на интегративного health-коуча - специалиста по модификации образа жизни и немедикаментозному оздоровлению.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Я помогаю своим клиентам вести здоровый образ жизни и предотвратить появление хронических заболеваний, а также повысить качество жизни, улучшить самочувствие, делюсь рекомендациями для восстановления и поддержания здоровья.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Я разбираю запрос клиента с разных сторон, что помогает увидеть полную картину конкретной ситуации и ее причины. Я использую различные подходы и инструменты, чтобы помочь проработать эти причины и придти к решению запроса клиента.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Одна из моих главных задач - осознание клиентом того, 
    <strong>
      что, как и почему нужно делать
    </strong>
    , чтобы достичь желаемого результата.
  </p>,
]
`;
